### DISCUSSION TITLE
Message protocol: Cucumber Messages

### DISCUSSION THREAD

**Discussion by bbros-dev** [cucumber-rs/gherkin#28] [General]
[Cucumber Messages](https://github.com/cucumber/cucumber/tree/master/messages) is a message protocol for representing results and other information from Cucumber - Replaces `json` and `junit` formatters.

Stability/Performance:

> With Cucumber Messages, several messages containing smaller pieces of information are emitted continuously to a stream,
> avoiding high memory consumption and enabling real-time processing of results.

Additional functionality (motivation):

> The protocol aims to decouple various components of the Cucumber platform so that:
>
>   - Each component only needs to know about a subset of messages
>   - Gherkin is decoupled from the Cucumber execution component
>       - This is part of a strategy to support other formats such as Markdown and Excel

**Comment by tyranron** [comment]
Moved this to https://github.com/cucumber-rs/cucumber/issues/169
